Add Best Practice: Naming Root Projects (#34468)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_general.adoc

=== Tags

<<tags_reference.adoc#tag:properties,`#properties`>>

[[name_your_root_project]]
== Name Your Root Project

Always name your root project in the `settings.gradle(.kts)` file.

=== Explanation

> While an empty `settings.gradle(.kts)` file is all that is minimally necessary for a multi-project build, you should always set the `rootProject.name` property in it.

Reviewer: Is the multi-project here needed? I understand this is also true for a build with only root project?

```suggestion
While just an empty `settings.gradle(.kts)` file is all you need, you should always set the `rootProject.name` property in it.
```
Rewrite with simpler language.

Without setting a name explicitly, the root project's name implicitly defaults to the name of the directory containing the build.
Some directory names may not be valid or suitable as project names, such as those containing spaces, Gradle logical path separators or other special characters.
The paths to tasks in build should be able to be specified reliably regardless of where the build lives.

> Explicitly setting the root project's name also ensures that it referred to consistently.

Reviewer: ```suggestion
Explicitly setting the root project's name also ensures that it is referred to consistently.
```
Or even rewrite for a simpler sentence 😅

Author: I'll wait for Laura for that - she's usually good at rewriting my too complex sentences.

Project names are used in various places, such as error messages, logs, and reports.
Builds may execute on a variety of machines or environments, such as CI servers, and should report the same root project name anywhere to make the project more comprehensible.

=== Example

==== Don't Do This

====
include::sample[dir="snippets/bestPractices/nameYourRootProject-avoid/kotlin",files="settings.gradle.kts[]"]
include::sample[dir="snippets/bestPractices/nameYourRootProject-avoid/groovy",files="settings.gradle[]"]
====

In this build, the settings file is empty and the root project is unnamed.

> If you try running the `projects` report, you will see an implicit name used for the root project that is based on the build's current location.

Reviewer: ```suggestion
If you try running the `projects` report, you will see a name based on the build's current location.
```
Rewrite for simpler english.
